product-backend/src/main/java/org/kainos/ea/db/JobRoleSingleViewDaoA.java

```diff
@@ -0,0 +1,31 @@
+package org.kainos.ea.db;
+
+
+import org.kainos.ea.cli.JobRoleSingleViewA;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class JobRoleSingleViewDaoA {
+
+    public JobRoleSingleViewA getJobRoleById(int id, Connection c) throws SQLException {
+        String getStatement = "SELECT * FROM `JobRole` WHERE `Id`=" + id + ";";
+        PreparedStatement st = c.prepareStatement(getStatement);
+        ResultSet rs = st.executeQuery();
+
+        while (rs.next()){
+            return new JobRoleSingleViewA(
+                    rs.getInt("Id"),
+                    rs.getString("Name"),
+                    rs.getString("Description"),
+                    rs.getString("SharePointLink")
+            );
+
+        }
+
+        return null;
+    }
```

Comment:
Use optionals
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html
```suggestion
    public Optional<JobRole> getJobRoleById(int id, Connection c) throws SQLException {
        String getStatement = "SELECT * FROM `JobRole` WHERE `Id`=" + id + ";";
        PreparedStatement st = c.prepareStatement(getStatement);
        ResultSet rs = st.executeQuery();

        while (rs.next()){
            return Optional.of(new JobRoleSingleViewA(
                    rs.getInt("Id"),
                    rs.getString("Name"),
                    rs.getString("Description"),
                    rs.getString("SharePointLink"))
            );

        }

        return Optional.empty();
    }
```

Service usage:

```
 Optional<JobRole> jobRole = jobRoleSingleViewDao.getJobRoleById(id, databaseConnector.getConnection());
return jobRole.orElseThrow(() -> new JobRoleDoesNotExistException());
```